### PR TITLE
ECMS-7508 making external content browsable

### PR DESCRIPTION
**Pull request by fmaatoug**

### CODE REVIEW

File: core/connector/src/main/java/org/exoplatform/wcm/connector/fckeditor/DriverConnector.java [hunk 1]
```diff
@@ -782,8 +782,10 @@ private boolean isDocument(Node node, String type) throws RepositoryException {
     TemplateService templateService = WCMCoreUtils.getService(TemplateService.class);
     List<String> documentTypeList = templateService.getDocumentTemplates();
     if (TYPE_EDITOR.equals(type)) {
-      documentTypeList.remove(NodetypeConstant.EXO_WEBCONTENT);
-      documentTypeList.remove(ILLUSTRATED_WEBCONTENT);
+        String[] browsableDocuments = System.getProperty(BROWSABLE_CONTENT).split(",");
```
Comment (wkhessairi) [inline core/connector/src/main/java/org/exoplatform/wcm/connector/fckeditor/DriverConnector.java]: What if BROWSABLE_CONTENT is not found ?, System.getProperty(BROWSABLE_CONTENT) returns null.

You should add a try catch here

Comment (wkhessairi) [inline core/connector/src/main/java/org/exoplatform/wcm/connector/fckeditor/DriverConnector.java]: or you can add a default value if it's null. And plz add a log warning on this expected parameter + add a pr to update exo.properties and default values in configuration.properties.

Comment (fmaatoug) [inline core/connector/src/main/java/org/exoplatform/wcm/connector/fckeditor/DriverConnector.java]: it cannot be NULL, it has a default value set in the property, please check the other pull request put in the jira comment

Comment (thomasdelhomenie) [inline core/connector/src/main/java/org/exoplatform/wcm/connector/fckeditor/DriverConnector.java]: It should use InitParams to get the parameter. It is more consistent and make tests easier.

Comment (thomasdelhomenie) [inline core/connector/src/main/java/org/exoplatform/wcm/connector/fckeditor/DriverConnector.java]: Agreed with @wkhessairi . We must always check for null to make sure the code is robust.

Comment (fmaatoug) [inline core/connector/src/main/java/org/exoplatform/wcm/connector/fckeditor/DriverConnector.java]: PR updated

File: core/connector/src/main/java/org/exoplatform/wcm/connector/fckeditor/DriverConnector.java [hunk 2]
```diff
@@ -134,6 +125,9 @@
   /** The file number limit on server side. */
   private int limitCountServer_ = 30;
 
+  private List<String> BROWSABLE_CONTENT =new ArrayList<String>();
```
Comment (thomasdelhomenie) [inline core/connector/src/main/java/org/exoplatform/wcm/connector/fckeditor/DriverConnector.java]: This is not a constant, it should not be in uppercase.

File: core/connector/src/main/java/org/exoplatform/wcm/connector/fckeditor/DriverConnector.java [hunk 3]
```diff
@@ -156,6 +150,9 @@ public DriverConnector(InitParams params) {
     if (params.getValueParam("upload.limit.count.server") != null) {
       limitCountServer_ = Integer.parseInt(params.getValueParam("upload.limit.count.server").getValue());
     }
+    if (params.getValueParam("exo.ecms.content.browsable") != null) {
+      BROWSABLE_CONTENT = Arrays.asList((params.getValueParam("exo.ecms.content.browsable").getValue()).split("\\s*,\\s*"));
```
Comment (thomasdelhomenie) [inline core/connector/src/main/java/org/exoplatform/wcm/connector/fckeditor/DriverConnector.java]: Why do you split on spaces ?

File: core/core-configuration/src/main/webapp/WEB-INF/conf/wcm-core/core-connector-configuration.xml [hunk 4]
```diff
@@ -97,6 +97,11 @@
         <description>Maximum number of the file uploading at a time</description>
         <value> ${wcm.connector.drives.serverLimit:20}</value>
       </value-param>
+      <value-param>
+        <name>exo.ecms.content.browsable</name>
+        <description>new browsable contents</description>
+        <value> ${exo.ecms.content.browsable:exo:webContent,exo:pictureOnHeadWebcontent}</value>
```
Comment (thomasdelhomenie) [inline core/core-configuration/src/main/webapp/WEB-INF/conf/wcm-core/core-connector-configuration.xml]: There is an leading space. It should be removed.

File: core/connector/src/main/java/org/exoplatform/wcm/connector/fckeditor/DriverConnector.java (hunk 2 above)
Comment (fmaatoug) [inline core/connector/src/main/java/org/exoplatform/wcm/connector/fckeditor/DriverConnector.java]: okay agreed

File: core/connector/src/main/java/org/exoplatform/wcm/connector/fckeditor/DriverConnector.java (hunk 3 above)
Comment (thomasdelhomenie) [inline core/connector/src/main/java/org/exoplatform/wcm/connector/fckeditor/DriverConnector.java]: Arf my bad, I totally missed the coma... :confused:

File: core/core-configuration/src/main/webapp/WEB-INF/conf/wcm-core/core-connector-configuration.xml (hunk 4 above)
Comment (fmaatoug) [inline core/core-configuration/src/main/webapp/WEB-INF/conf/wcm-core/core-connector-configuration.xml]: done

File: core/connector/src/main/java/org/exoplatform/wcm/connector/fckeditor/DriverConnector.java (hunk 2 above)
Comment (fmaatoug) [inline core/connector/src/main/java/org/exoplatform/wcm/connector/fckeditor/DriverConnector.java]: done

File: core/connector/src/main/java/org/exoplatform/wcm/connector/fckeditor/DriverConnector.java [hunk 5]
```diff
@@ -782,8 +779,10 @@ private boolean isDocument(Node node, String type) throws RepositoryException {
     TemplateService templateService = WCMCoreUtils.getService(TemplateService.class);
     List<String> documentTypeList = templateService.getDocumentTemplates();
     if (TYPE_EDITOR.equals(type)) {
-      documentTypeList.remove(NodetypeConstant.EXO_WEBCONTENT);
-      documentTypeList.remove(ILLUSTRATED_WEBCONTENT);
+      List<String> browsableDocuments = browsableContent;
```
Comment (thomasdelhomenie) [inline core/connector/src/main/java/org/exoplatform/wcm/connector/fckeditor/DriverConnector.java]: Looks useless to assign to a new variable.

File: core/connector/src/main/java/org/exoplatform/wcm/connector/fckeditor/DriverConnector.java [hunk 6]
```diff
@@ -782,8 +779,10 @@ private boolean isDocument(Node node, String type) throws RepositoryException {
...
+      for (String browsableDocument : browsableDocuments) {
```
Comment (thomasdelhomenie) [inline core/connector/src/main/java/org/exoplatform/wcm/connector/fckeditor/DriverConnector.java]: We should check for browsableContent nullity before iterating on it, for robutness.

Comment (fmaatoug) [inline core/connector/src/main/java/org/exoplatform/wcm/connector/fckeditor/DriverConnector.java]: it's already checked in initParams should i redo the test in this method ?

File: core/connector/src/main/java/org/exoplatform/wcm/connector/fckeditor/DriverConnector.java (hunk 5 above)
Comment (fmaatoug) [inline core/connector/src/main/java/org/exoplatform/wcm/connector/fckeditor/DriverConnector.java]: agreed

File: core/connector/src/main/java/org/exoplatform/wcm/connector/fckeditor/DriverConnector.java (hunk 6 above)
Comment (fmaatoug) [inline core/connector/src/main/java/org/exoplatform/wcm/connector/fckeditor/DriverConnector.java]: the nullity of this variable is already checked in initParams() should i check it again ?

Comment (fmaatoug) [inline core/connector/src/main/java/org/exoplatform/wcm/connector/fckeditor/DriverConnector.java]: agreed

File: core/connector/src/main/java/org/exoplatform/wcm/connector/fckeditor/DriverConnector.java (hunk 5 above)
Comment (fmaatoug) [inline core/connector/src/main/java/org/exoplatform/wcm/connector/fckeditor/DriverConnector.java]: done

File: core/connector/src/main/java/org/exoplatform/wcm/connector/fckeditor/DriverConnector.java (hunk 6 above)
Comment (fmaatoug) [inline core/connector/src/main/java/org/exoplatform/wcm/connector/fckeditor/DriverConnector.java]: done

**Comment by thomasdelhomenie** [comment]
@wkhessairi is that ok for me ?

**Comment by wkhessairi** [comment]
@thomasdelhomenie yes LGTM

**Pull request merged**